### ISSUE TITLE
"Connecting..." tab text \ FF33 \ v1.5.28.1

### ISSUE THREAD

**Issue by irvinm** [ithinc/tabutils#126]
After fully loading a tab, sometimes the text remaining on the tab is just "Connecting..." vs. the text provided by the website.  Reloading the tab sometimes corrects the problem, but sometimes goes back to "Connecting..." after a period of time.  (Assuming a dynamically loading page refreshes information)

This is obviously problematic when restoring multiple tabs and some % of them just have "Connecting..." as their tab text.

This does not happen for 100% of the tabs I load, but often enough that is it problematic.  (You can only identify what is on that tab by the website icon IF it is provided, else you get a blank icon with only "Connecting...")

My comments on the Firefox TU forum:
http://forums.mozillazine.org/viewtopic.php?f=48&t=1702465&p=13838153#p13838153

Another comment on the Firefox TU forum:
http://forums.mozillazine.org/viewtopic.php?f=48&t=1702465&p=13824053#p13824053

**Comment by RedHorizon** [comment]
I found reproduce steps!
FF33, TU1.5.28.1
Create a new FF profile, launch FF, install TU using Tools - Addons
Firefox will restart, there will be https://www.mozilla.org/ru/firefox/33.0.1/firstrun/ page still open - switch to this tab, type yandex.ru in url bar, press enter and immediately after that open new tab using new tab button on tab bar
Wait 5-10 seconds - title will change from "Yandex" to "Connecting"

PS Yandex.ru is russian google.
http://en.wikipedia.org/wiki/Yandex

**Comment by irvinm** [comment]
Excellent!  That will certainly help debug the issue when the time comes ...

**Comment by koscos** [comment]
FF33.0.2 TU1.6pre21 

Same issue too...

**Comment by koscos** [comment]
this issue may be caused by Twitter widgets

**Comment by irvinm** [comment]
Not in my case as I don't have anything Twitter related.

**Comment by photonometric** [comment]
Indeed, I'm getting this effect on a fresh portable install of FF 34.0.5, with only Tree Style Tabs running. Also nothing twitter-related on my computer.

**Comment by awalkonthebeach** [comment]
Oh man this "Connecting..." crap on the tabs is so frikkin annoying!  I'm keeping TU off until this is fixed.  Pffft

**Comment by photonometric** [comment]
This seems to have stopped happening to me since running FF 35....been using TU quite a bit and is running a whole lot smoother in general.

**Comment by yfdyh000** [comment]
I cannot reproduce the problem with Fx33 or Fx36 + TU.
The http://www.yandex.ru/ will delayed loading a http://yastatic.net/browser-cookie/_fc.js?_= ..., it will resume the original title if the load is completed.

**Comment by irvinm** [comment]
I am still seeing several examples where the page loads completely and the tab text is still "Connecting ...".  Even in my current session "Facebook", "Google", and others.  I will see I can also take a clean profile and reproduce the problem.

**Comment by irvinm** [comment]
Ok, I loaded up FF 35.0.1 with no add-ons except for TU 1.5.28.1.

i opened up yahoo.com and just started right clicking links with "Open link in new tab".  All of the links worked from yahoo (tried 5) off the main page, then I did a search for "google" and did the same thing with those results.  The first 3 loaded fine, then the next 3 have "Connecting..." as the text.

Session at the end looks like:

https://www.yahoo.com/
https://www.yahoo.com/style
https://www.yahoo.com/beauty
https://www.yahoo.com/food
https://www.yahoo.com/movies
https://www.yahoo.com/travel
https://www.yahoo.com/tech
https://www.facebook.com/
https://search.yahoo.com/yhs/search?p=google&ei=UTF-8&hspart=mozilla&hsimp=yhs-001
http://en.wikipedia.org/wiki/Google
https://www.facebook.com/Google
http://news.yahoo.com/sony-releases-google-glass-alternative-175234008--abc-news-tech.html;_ylt=A0LEVjWinONU5X8ACCgnnIlQ
http://finance.yahoo.com/news/google-employees-chattering-huge-threat-171327251.html;_ylt=A0LEVjWinONU5X8ACignnIlQ
http://finance.yahoo.com/q;_ylt=A0LEVjWinONU5X8ADCgnnIlQ;_ylu=X3oDMTEzOWJvMDNsBHNlYwNzcgRwb3MDNARjb2xvA2JmMQR2dGlkA1lIUzAwNF8x?s=GOOGL

So ... 14 tabs, 3 initial failures.  I was able to do "reloads" of those last 3 pages and eventually got the titles to load.  However, some reverted back to "Connecting...".  Obviously some dynamically loading or refreshing going on.

**Comment by yfdyh000** [comment]
Ah, I think I see the issue if the tabs loading failed in Fx35 with the original TU 1.5.28.1.
I don't know the cause of it so far, but I use my own fixes version does not seen the problem, so please try the: https://github.com/yfdyh000/tabutils/releases

**Comment by irvinm** [comment]
I tried https://github.com/yfdyh000/tabutils/releases/download/2015.02.18/tabutils-Fixed-2015.02.18.xpi and things are looking pretty good.  No "Connecting..." tabs so far.  Let me kick the tires before closing.

**Comment by yfdyh000** [comment]
Updates: I guess the https://github.com/yfdyh000/tabutils/commit/69136626d66da4b4d74999c00acd5211db856885 commit fixes it, because the setAttribute be break, which includes set the title of tabs.

**Comment by irvinm** [comment]
This looks to be fixed.  Woot!  Do we want to close this issue or leave it open until it gets merged back into trunk by ithinc?

**Comment by yfdyh000** [comment]
I recommend keeping it open for others to reference and future merger, although the original author seems to shelve the project so far.

**Comment by photonometric** [comment]
I was premature in saying this had stopped in FF35...I had several hundred tabs open and never saw it in any of the loaded tabs, but after closing all and only working with about ten, I see it in about a third. 

The release @yfdyh000 linked to fixes it nicely...thanks a lot for your work! ^^
